src/extent.c

```diff
@@ -2279,3 +2279,42 @@ extent_boot(void) {
 
 	return false;
 }
+
+void
+extent_util_stats_get(tsdn_t *tsdn, const void *ptr, size_t *nfree,
+    size_t *nregs, size_t *size, void **slabcur_addr) {
+	*nfree = *nregs = *size = 0;
+	if (slabcur_addr != NULL) {
+		*slabcur_addr = NULL;
+	}
+
+	const extent_t *extent = iealloc(tsdn, ptr);
+	if (unlikely(extent == NULL)) {
+		return;
+	}
+
+	*size = extent_size_get(extent);
+	if (!extent_slab_get(extent)) {
+		*nregs = 1;
+		return;
+	}
+
+	*nfree = extent_nfree_get(extent);
+	const szind_t szind = extent_szind_get(extent);
+	*nregs = bin_infos[szind].nregs;
+	assert(*nfree <= *nregs);
+	assert(*nfree * extent_usize_get(extent) <= *size);
+
+	if (slabcur_addr != NULL) {
+		const arena_t *arena = extent_arena_get(extent);
+		assert(arena != NULL);
+		const unsigned binshard = extent_binshard_get(extent);
+		bin_t *bin = &arena->bins[szind].bin_shards[binshard];
+
+		malloc_mutex_lock(tsdn, &bin->lock);
+		const extent_t *cur = bin->slabcur;
+		*slabcur_addr = extent_addr_get(cur);
+		assert(*slabcur_addr != NULL);
+		malloc_mutex_unlock(tsdn, &bin->lock);
+	}
+}
```

Comment:
... and in the end, have a `label_return` (after the normal return) which does
```
        if (slabcur_addr != NULL) {
		*slabcur_addr = NULL;
	}
        return;
```

With this refactor, the values are set exactly once for each case, and it may enable the compiler to warn us if some values are missing initialization.